src/general/AutoModLoader.cs

```diff
@@ -0,0 +1,65 @@
+using System.IO;
+using System.Reflection;
+using Godot;
+using Newtonsoft.Json;
+using Directory = System.IO.Directory;
+using File = System.IO.File;
+
+public class AutoModLoader : Node
+{
+    private AutoModLoader()
+    {
+        DirectoryInfo modFolder = Directory.CreateDirectory($"{Directory.GetCurrentDirectory()}\\mods");
+        foreach (DirectoryInfo currentMod in modFolder.EnumerateDirectories())
+        {
+            if (!File.Exists(currentMod.FullName + "/mod_info.json"))
+            {
+                continue;
+            }
+
+            var currentModInfo =
+                JsonConvert.DeserializeObject<ModInfo>(ReadJSONFile(currentMod.FullName + "/mod_info.json"));
+            if (currentModInfo.AutoLoad)
+            {
+                if (string.IsNullOrEmpty(currentModInfo.Dll))
+                {
+                    if (File.Exists(currentMod.FullName + "/" + currentModInfo.Dll))
+                    {
+                        Assembly.LoadFile(currentMod.FullName + "/" + currentModInfo.Dll);
```

Comment:
Allowing to load the assembly from within a pck would be nice, similarly to how the main game dll is included in Thrive.pck

Reply:
i am not sure how to really do that as LoadFile uses a path and you can't use a godot path with it

Reply:
It seems that Assembly.Load (https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.load?view=netcore-3.1) might work if the raw data is read with godot and then passed there.
Another option would be to ask Godot to translate a godot path to an absolute path and then proceed.